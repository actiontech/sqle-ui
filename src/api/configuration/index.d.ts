import {
  IGetDriversResV1,
  IGetLDAPConfigurationResV1,
  ILDAPConfigurationReqV1,
  IBaseRes,
  IGetSMTPConfigurationResV1,
  IUpdateSMTPConfigurationReqV1,
  IGetSystemVariablesResV1,
  IUpdateSystemVariablesReqV1
} from '../common.d';

export interface IGetDriversV1Return extends IGetDriversResV1 {}

export interface IGetLDAPConfigurationV1Return
  extends IGetLDAPConfigurationResV1 {}

export interface IUpdateLDAPConfigurationV1Params
  extends ILDAPConfigurationReqV1 {}

export interface IUpdateLDAPConfigurationV1Return extends IBaseRes {}

export interface IGetSMTPConfigurationV1Return
  extends IGetSMTPConfigurationResV1 {}

export interface IUpdateSMTPConfigurationV1Params
  extends IUpdateSMTPConfigurationReqV1 {}

export interface IUpdateSMTPConfigurationV1Return extends IBaseRes {}

export interface IGetSystemVariablesV1Return extends IGetSystemVariablesResV1 {}

export interface IUpdateSystemVariablesV1Params
  extends IUpdateSystemVariablesReqV1 {}

export interface IUpdateSystemVariablesV1Return extends IBaseRes {}
