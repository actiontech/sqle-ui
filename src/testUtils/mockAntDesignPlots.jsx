import { cloneDeep, isObject } from 'lodash';
const MockPlots = (props) => {
  const cloneProps = cloneDeep(props);

  Object.keys(cloneProps).forEach((key) => {
    if (isObject(cloneProps[key])) {
      cloneProps[key] = JSON.stringify(cloneProps[key]);
    }
  });
  return <div {...cloneProps} />;
};

const Line = MockPlots;
const Pie = MockPlots;
const RadialBar = MockPlots;
const Gauge = MockPlots;
const Column = MockPlots;
const mockRegisterShape = jest.fn();
const G2 = {
  registerShape: jest.fn(),
};

export { Line, Pie, RadialBar, Gauge, Column, G2, mockRegisterShape };

// eslint-disable-next-line import/no-anonymous-default-export
export default {
  Line,
  Pie,
  RadialBar,
  Gauge,
  Column,
  G2,
};
