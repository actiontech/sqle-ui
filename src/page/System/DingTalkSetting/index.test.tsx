import { fireEvent, render, screen, act } from '@testing-library/react';
import DingTalkSetting from '.';
import configuration from '../../../api/configuration';
import { resolveThreeSecond } from '../../../testUtils/mockRequest';

describe('test DingTalkSetting', () => {
  let getDingTalkConfigSpy: jest.SpyInstance;
  let updateDingTalkConfigSpy: jest.SpyInstance;
  let testDingTalkConfigSpy: jest.SpyInstance;
  beforeEach(() => {
    getDingTalkConfigSpy = mockGetDingTalkConfiguration();
    updateDingTalkConfigSpy = mockUpdateDingTalkConfiguration();
    testDingTalkConfigSpy = mockTestDingTalkConfiguration();
    jest.useFakeTimers();
  });
  afterEach(() => {
    jest.clearAllMocks();
    jest.clearAllTimers();
    jest.useRealTimers();
  });

  const mockGetDingTalkConfiguration = () => {
    const spy = jest.spyOn(configuration, 'getDingTalkConfigurationV1');
    spy.mockImplementation(() =>
      resolveThreeSecond({
        app_key: 'app_key',
        app_secret: 'app_secret',
        is_enable_ding_talk_notify: false,
      })
    );
    return spy;
  };

  const mockTestDingTalkConfiguration = () => {
    const spy = jest.spyOn(configuration, 'testDingTalkConfigV1');
    spy.mockImplementation(() =>
      resolveThreeSecond({
        is_ding_talk_send_normal: true,
      })
    );
    return spy;
  };

  const mockUpdateDingTalkConfiguration = () => {
    const spy = jest.spyOn(configuration, 'updateDingTalkConfigurationV1');
    spy.mockImplementation(() => resolveThreeSecond({}));
    return spy;
  };

  test('should match snapshot', async () => {
    const { container } = render(<DingTalkSetting />);
    expect(getDingTalkConfigSpy).toBeCalledTimes(1);

    await act(async () => jest.advanceTimersByTime(3000));

    expect(container).toMatchSnapshot();

    fireEvent.click(screen.getByText('common.modify'));
    expect(container).toMatchSnapshot();
  });

  test('should match snapshot when open dingTalk configuration', async () => {
    getDingTalkConfigSpy.mockImplementation(() =>
      resolveThreeSecond({
        app_key: 'app_key',
        app_secret: 'app_secret',
        is_enable_ding_talk_notify: true,
      })
    );
    const { container } = render(<DingTalkSetting />);
    expect(getDingTalkConfigSpy).toBeCalledTimes(1);

    await act(async () => jest.advanceTimersByTime(3000));

    expect(container).toMatchSnapshot();

    fireEvent.click(screen.getByText('common.modify'));
    expect(container).toMatchSnapshot();
  });

  test('should be able to update dingTalk configuration', async () => {
    render(<DingTalkSetting />);

    await act(async () => jest.advanceTimersByTime(3000));

    fireEvent.click(screen.getByText('common.modify'));
    fireEvent.click(screen.getByLabelText('system.dingTalk.enable'));
    fireEvent.change(screen.getByLabelText('AppKey'), {
      target: { value: 'update-appKey' },
    });
    fireEvent.change(screen.getByLabelText('AppSecret'), {
      target: { value: 'update-appSecret' },
    });
    expect(updateDingTalkConfigSpy).toBeCalledTimes(0);

    fireEvent.click(screen.getByText('common.submit'));
    await act(async () => jest.advanceTimersByTime(0));

    expect(updateDingTalkConfigSpy).toBeCalledTimes(1);
    expect(updateDingTalkConfigSpy).toBeCalledWith({
      is_enable_ding_talk_notify: true,
      app_key: 'update-appKey',
      app_secret: 'update-appSecret',
    });
    expect(screen.getByText('common.submit').closest('button')).toHaveClass(
      'ant-btn-loading'
    );

    expect(screen.getByText('common.cancel').closest('button')).toBeDisabled();

    await act(async () => jest.advanceTimersByTime(3000));

    expect(screen.getByText('common.submit').closest('button')).not.toHaveClass(
      'ant-btn-loading'
    );

    expect(
      screen.getByText('common.cancel').closest('button')
    ).not.toBeDisabled();

    expect(screen.getByLabelText('AppKey')).toHaveValue('');
    expect(screen.getByLabelText('AppSecret')).toHaveValue('');
    expect(getDingTalkConfigSpy).toBeCalledTimes(2);
    expect(screen.getByText('common.modify')).toBeInTheDocument();
  });

  test('should send request when clicking test DingTalk button', async () => {
    render(<DingTalkSetting />);
    expect(testDingTalkConfigSpy).toBeCalledTimes(0);

    await act(async () => jest.advanceTimersByTime(3000));

    fireEvent.click(screen.getByText('system.dingTalk.test'));

    expect(testDingTalkConfigSpy).toBeCalledTimes(1);

    await act(async () => jest.advanceTimersByTime(3000));

    expect(screen.getByText('system.dingTalk.testSuccess')).toBeInTheDocument();

    await act(async () => jest.advanceTimersByTime(3000));

    expect(
      screen.queryByText('system.dingTalk.testSuccess')
    ).not.toBeInTheDocument();

    jest.clearAllMocks();
    testDingTalkConfigSpy.mockImplementation(() => {
      return resolveThreeSecond({
        is_ding_talk_send_normal: false,
        send_error_message: 'error message',
      });
    });
    fireEvent.click(screen.getByText('system.dingTalk.test'));
    await act(async () => jest.advanceTimersByTime(3000));

    expect(screen.getByText('error message')).toBeInTheDocument();

    await act(async () => jest.advanceTimersByTime(3000));

    expect(screen.queryByText('error message')).not.toBeInTheDocument();
  });
});
