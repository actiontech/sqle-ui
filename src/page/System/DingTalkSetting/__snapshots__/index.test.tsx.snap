// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test DingTalkSetting should match snapshot 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          system.title.dingTalk
          <div
            class="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              class="ant-space-item"
            >
              <span
                aria-label="info-circle"
                class="anticon anticon-info-circle text-orange"
                role="img"
                style="font-size: 14px; margin-left: 6px;"
              >
                <svg
                  aria-hidden="true"
                  data-icon="info-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                  />
                  <path
                    d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <section>
        <div
          class="ant-descriptions"
        >
          <div
            class="ant-descriptions-view"
          >
            <table>
              <tbody>
                <tr
                  class="ant-descriptions-row"
                >
                  <td
                    class="ant-descriptions-item"
                    colspan="3"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-label"
                      >
                        system.dingTalk.enable
                      </span>
                      <span
                        class="ant-descriptions-item-content"
                      >
                        common.close
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  class="ant-descriptions-row"
                >
                  <td
                    class="ant-descriptions-item"
                    colspan="3"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-label"
                      >
                        AppKey
                      </span>
                      <span
                        class="ant-descriptions-item-content"
                      >
                        <div
                          class="ant-typography"
                        >
                          app_key
                        </div>
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  class="ant-descriptions-row"
                >
                  <td
                    class="ant-descriptions-item"
                    colspan="3"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-content"
                      >
                        <button
                          class="ant-btn ant-btn-primary"
                          type="button"
                        >
                          <span>
                            common.modify
                          </span>
                        </button>
                      </span>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </section>
      <form
        class="ant-form ant-form-horizontal"
        hidden=""
      >
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
          >
            <label
              class=""
              for="enabled"
              title="system.dingTalk.enable"
            >
              system.dingTalk.enable
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <button
                  aria-checked="false"
                  class="ant-switch"
                  id="enabled"
                  role="switch"
                  type="button"
                >
                  <div
                    class="ant-switch-handle"
                  />
                  <span
                    class="ant-switch-inner"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
          >
            <label
              class="ant-form-item-required"
              for="appKey"
              title="AppKey"
            >
              AppKey
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  id="appKey"
                  placeholder="common.form.placeholder.input"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
          >
            <label
              class="ant-form-item-required"
              for="appSecret"
              title="AppSecret"
            >
              AppSecret
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  id="appSecret"
                  placeholder="common.form.placeholder.input"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
          >
            <label
              class="ant-form-item-no-colon"
              title=" "
            >
               
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-space ant-space-horizontal ant-space-align-center"
                >
                  <div
                    class="ant-space-item"
                    style="margin-right: 8px;"
                  >
                    <button
                      class="ant-btn ant-btn-primary"
                      type="submit"
                    >
                      <span>
                        common.submit
                      </span>
                    </button>
                  </div>
                  <div
                    class="ant-space-item"
                  >
                    <button
                      class="ant-btn"
                      type="button"
                    >
                      <span>
                        common.cancel
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`test DingTalkSetting should match snapshot 2`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          system.title.dingTalk
          <div
            class="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              class="ant-space-item"
            >
              <span
                aria-label="info-circle"
                class="anticon anticon-info-circle text-orange"
                role="img"
                style="font-size: 14px; margin-left: 6px;"
              >
                <svg
                  aria-hidden="true"
                  data-icon="info-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                  />
                  <path
                    d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <section
        hidden=""
      >
        <div
          class="ant-descriptions"
        >
          <div
            class="ant-descriptions-view"
          >
            <table>
              <tbody>
                <tr
                  class="ant-descriptions-row"
                >
                  <td
                    class="ant-descriptions-item"
                    colspan="3"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-label"
                      >
                        system.dingTalk.enable
                      </span>
                      <span
                        class="ant-descriptions-item-content"
                      >
                        common.close
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  class="ant-descriptions-row"
                >
                  <td
                    class="ant-descriptions-item"
                    colspan="3"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-label"
                      >
                        AppKey
                      </span>
                      <span
                        class="ant-descriptions-item-content"
                      >
                        <div
                          class="ant-typography"
                        >
                          app_key
                        </div>
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  class="ant-descriptions-row"
                >
                  <td
                    class="ant-descriptions-item"
                    colspan="3"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-content"
                      >
                        <button
                          ant-click-animating-without-extra-node="false"
                          class="ant-btn ant-btn-primary"
                          type="button"
                        >
                          <span>
                            common.modify
                          </span>
                        </button>
                      </span>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </section>
      <form
        class="ant-form ant-form-horizontal"
      >
        <div
          class="ant-row ant-form-item ant-form-item-has-success"
        >
          <div
            class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
          >
            <label
              class=""
              for="enabled"
              title="system.dingTalk.enable"
            >
              system.dingTalk.enable
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <button
                  aria-checked="false"
                  class="ant-switch"
                  id="enabled"
                  role="switch"
                  type="button"
                >
                  <div
                    class="ant-switch-handle"
                  />
                  <span
                    class="ant-switch-inner"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item ant-form-item-has-success"
        >
          <div
            class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
          >
            <label
              class="ant-form-item-required"
              for="appKey"
              title="AppKey"
            >
              AppKey
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  id="appKey"
                  placeholder="common.form.placeholder.input"
                  type="text"
                  value="app_key"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item ant-form-item-has-success"
        >
          <div
            class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
          >
            <label
              class="ant-form-item-required"
              for="appSecret"
              title="AppSecret"
            >
              AppSecret
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  id="appSecret"
                  placeholder="common.form.placeholder.input"
                  type="text"
                  value="app_secret"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
          >
            <label
              class="ant-form-item-no-colon"
              title=" "
            >
               
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-space ant-space-horizontal ant-space-align-center"
                >
                  <div
                    class="ant-space-item"
                    style="margin-right: 8px;"
                  >
                    <button
                      class="ant-btn ant-btn-primary"
                      type="submit"
                    >
                      <span>
                        common.submit
                      </span>
                    </button>
                  </div>
                  <div
                    class="ant-space-item"
                  >
                    <button
                      class="ant-btn"
                      type="button"
                    >
                      <span>
                        common.cancel
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`test DingTalkSetting should match snapshot when open dingTalk configuration 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          system.title.dingTalk
          <div
            class="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              class="ant-space-item"
            >
              <span
                aria-label="info-circle"
                class="anticon anticon-info-circle text-orange"
                role="img"
                style="font-size: 14px; margin-left: 6px;"
              >
                <svg
                  aria-hidden="true"
                  data-icon="info-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                  />
                  <path
                    d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <section>
        <div
          class="ant-descriptions"
        >
          <div
            class="ant-descriptions-view"
          >
            <table>
              <tbody>
                <tr
                  class="ant-descriptions-row"
                >
                  <td
                    class="ant-descriptions-item"
                    colspan="3"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-label"
                      >
                        system.dingTalk.enable
                      </span>
                      <span
                        class="ant-descriptions-item-content"
                      >
                        common.open
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  class="ant-descriptions-row"
                >
                  <td
                    class="ant-descriptions-item"
                    colspan="3"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-label"
                      >
                        AppKey
                      </span>
                      <span
                        class="ant-descriptions-item-content"
                      >
                        <div
                          class="ant-typography"
                        >
                          app_key
                        </div>
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  class="ant-descriptions-row"
                >
                  <td
                    class="ant-descriptions-item"
                    colspan="3"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-content"
                      >
                        <button
                          class="ant-btn ant-btn-primary"
                          type="button"
                        >
                          <span>
                            common.modify
                          </span>
                        </button>
                      </span>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </section>
      <form
        class="ant-form ant-form-horizontal"
        hidden=""
      >
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
          >
            <label
              class=""
              for="enabled"
              title="system.dingTalk.enable"
            >
              system.dingTalk.enable
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <button
                  aria-checked="false"
                  class="ant-switch"
                  id="enabled"
                  role="switch"
                  type="button"
                >
                  <div
                    class="ant-switch-handle"
                  />
                  <span
                    class="ant-switch-inner"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
          >
            <label
              class="ant-form-item-required"
              for="appKey"
              title="AppKey"
            >
              AppKey
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  id="appKey"
                  placeholder="common.form.placeholder.input"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
          >
            <label
              class="ant-form-item-required"
              for="appSecret"
              title="AppSecret"
            >
              AppSecret
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  id="appSecret"
                  placeholder="common.form.placeholder.input"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
          >
            <label
              class="ant-form-item-no-colon"
              title=" "
            >
               
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-space ant-space-horizontal ant-space-align-center"
                >
                  <div
                    class="ant-space-item"
                    style="margin-right: 8px;"
                  >
                    <button
                      class="ant-btn ant-btn-primary"
                      type="submit"
                    >
                      <span>
                        common.submit
                      </span>
                    </button>
                  </div>
                  <div
                    class="ant-space-item"
                  >
                    <button
                      class="ant-btn"
                      type="button"
                    >
                      <span>
                        common.cancel
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`test DingTalkSetting should match snapshot when open dingTalk configuration 2`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          system.title.dingTalk
          <div
            class="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              class="ant-space-item"
            >
              <span
                aria-label="info-circle"
                class="anticon anticon-info-circle text-orange"
                role="img"
                style="font-size: 14px; margin-left: 6px;"
              >
                <svg
                  aria-hidden="true"
                  data-icon="info-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                  />
                  <path
                    d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <section
        hidden=""
      >
        <div
          class="ant-descriptions"
        >
          <div
            class="ant-descriptions-view"
          >
            <table>
              <tbody>
                <tr
                  class="ant-descriptions-row"
                >
                  <td
                    class="ant-descriptions-item"
                    colspan="3"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-label"
                      >
                        system.dingTalk.enable
                      </span>
                      <span
                        class="ant-descriptions-item-content"
                      >
                        common.open
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  class="ant-descriptions-row"
                >
                  <td
                    class="ant-descriptions-item"
                    colspan="3"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-label"
                      >
                        AppKey
                      </span>
                      <span
                        class="ant-descriptions-item-content"
                      >
                        <div
                          class="ant-typography"
                        >
                          app_key
                        </div>
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  class="ant-descriptions-row"
                >
                  <td
                    class="ant-descriptions-item"
                    colspan="3"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-content"
                      >
                        <button
                          ant-click-animating-without-extra-node="false"
                          class="ant-btn ant-btn-primary"
                          type="button"
                        >
                          <span>
                            common.modify
                          </span>
                        </button>
                      </span>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </section>
      <form
        class="ant-form ant-form-horizontal"
      >
        <div
          class="ant-row ant-form-item ant-form-item-has-success"
        >
          <div
            class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
          >
            <label
              class=""
              for="enabled"
              title="system.dingTalk.enable"
            >
              system.dingTalk.enable
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <button
                  aria-checked="true"
                  class="ant-switch ant-switch-checked"
                  id="enabled"
                  role="switch"
                  type="button"
                >
                  <div
                    class="ant-switch-handle"
                  />
                  <span
                    class="ant-switch-inner"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item ant-form-item-has-success"
        >
          <div
            class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
          >
            <label
              class="ant-form-item-required"
              for="appKey"
              title="AppKey"
            >
              AppKey
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  id="appKey"
                  placeholder="common.form.placeholder.input"
                  type="text"
                  value="app_key"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item ant-form-item-has-success"
        >
          <div
            class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
          >
            <label
              class="ant-form-item-required"
              for="appSecret"
              title="AppSecret"
            >
              AppSecret
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  id="appSecret"
                  placeholder="common.form.placeholder.input"
                  type="text"
                  value="app_secret"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
          >
            <label
              class="ant-form-item-no-colon"
              title=" "
            >
               
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-space ant-space-horizontal ant-space-align-center"
                >
                  <div
                    class="ant-space-item"
                    style="margin-right: 8px;"
                  >
                    <button
                      class="ant-btn ant-btn-primary"
                      type="submit"
                    >
                      <span>
                        common.submit
                      </span>
                    </button>
                  </div>
                  <div
                    class="ant-space-item"
                  >
                    <button
                      class="ant-btn"
                      type="button"
                    >
                      <span>
                        common.cancel
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
